test/unit/transport.test.js

```diff
@@ -1633,12 +1633,13 @@ test('Should cast to boolean HEAD request', t => {
       path: '/400'
     }, (err, { body, statusCode }) => {
       t.ok(err instanceof ResponseError)
+      t.false(typeof err.body === 'boolean')
```

Comment:
shouldn't we test `body` contains `payload`?

Reply:
It should never contain `payload` 🤔 
Also, if case of HEAD request, the body should be either a boolean or something else :)